docs/flags.md

```diff
@@ -0,0 +1,206 @@
+
+You want to be able to write a doc that tells users to run `tilt up consumer` the first time and know that's enough that they'll be set up. Running Tilt subsequent times without arguments should default to the mode they already have set.
+
+#### Command-Lines
+* `tilt up`: run all services
+* `tilt up consumer`: run consumer services
+
+#### Tiltfile
+```python
+config.define_string_list("to-run", args=True)
+cfg = config.parse()
+
+groups = {
+  'consumer': ['a', 'b', 'c'],
+  'enterprise': ['a', 'b', 'd'],
+}
+
+resources = []
+for arg in cfg['args']:
+  if arg in groups:
+    resources += groups[arg]
+  else:
+    resources.append(r)
+config.set_resources(resources)
+```
+
+### Specify services to edit
+You have many services that your users frequently need to run, but don't expect to edit. Your YAML includes good enough images. (Perhaps they're from a suitably recent CI invocation, or refer to standard images.)
+
+#### Command-Lines
+* `tilt up`: run all services, editing none
+* `tilt up -- --to-edit b`: run all services, editing b
+* `tilt up -- consumer --to-edit b --to-edit c`: run consumer services and setup b and c for editing.
+
+#### Tiltfile
+```python
+config.define_string_list("to-run", args=True)
+config.define_string_list("to-edit")
+cfg = config.parse()
+
+# omitting the code from above Tiltfile that includes correct resources
+
+# Only configure the build for services we want to edit
+# Thanks to Tilt's existing behavior, an image that is mentioned in YAML but has no build set
+# will use the existing tag.
+if 'a' in cfg['to-edit']:
+  docker_build('a', './a')
+if 'b' in cfg['to-edit']:
+  docker_build('b', './b')
+if 'c' in cfg['to-edit']:
+  docker_build('c', './c')
+if 'd' in cfg['to-edit']:
+  docker_build('d', './d')
+```
+
+## How Options Work
+Tilt stores options in a file next to the Tiltfile called `tilt-config.json`. When the Tiltfile executes `config.parse()`, it loads that file as JSON and merges the current args.
+
+Any change to the flags file will cause Tilt to reexecute the Tiltfile. You can modify the flags file with an editor or write scripts that write values into it, and Tilt will update.
+
+A call to `tilt args` will connect to the running Tilt instance on the server port and cause Tilt to reexecute the Tiltfile.
+
+### Comparison to Default Behavior
+If you don't call `config.parse`, Tilt's default behavior is to set resources to any (non-empty) passed args. The equivalent of a Tiltfile:
+```python
+config.define_string_list("args", args=True)
+cfg = config.parse()
+if cfg['args']:
+  config.set_resources(cfg['args'])
+```
+
+## Future Directions
+This section describes places we expect flags to go. Let us know if any of these would be particularly helpful to you and your team.
+
```

Comment:
do you want to pitch stickiness here?

Reply:
I'd be very very content if we just had "store args in file instead of in memory", which I think is a good experience and simpler.

Reply:
maybe my question should be "is there something in that vein on which you'd like to solicit feedback here?"